src/transformers/pipelines/base.py

```diff
@@ -294,8 +294,35 @@ def infer_framework_load_model(
                     model = model.eval()
                 # Stop loading on the first successful load.
                 break
-            except (OSError, ValueError):
-                all_traceback[model_class.__name__] = traceback.format_exc()
+            except (OSError, ValueError, TypeError, RuntimeError):
```

Comment:
I'm assuming `TypeError` / `RuntimeError` are the exceptions thrown? (I can't find references, it would be nice to add a comment here with some info)

Reply:
Added a quick comment about it, but essentially the errors are something like this:
```
e.g. RuntimeError: "bfloat16" isn’t supported on CUDA device 0
e.g. RuntimeError: Expected scalar type BFloat16 but found Float

```